`make misspell` doesn't show mispellings

## Bug Report

If you add a misspelling (like this mispelling of compatibility in the `unmanaged-cluster` plugin):
```go
log.Style(outputIndent, color.FgYellow).ReplaceLinef("Reading compatability ProviderConfiguration ... ")
```

And then running `make misspell` fails _but_ does not show where it failed:
```shell
❯ make misspell
hack/check/check-misspell.sh
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  8851  100  8851    0     0  29503      0 --:--:-- --:--:-- --:--:-- 29503
client9/misspell INFO checking GitHub for latest tag
client9/misspell INFO found version: 0.3.4 for v0.3.4/linux/64bit
client9/misspell DEBUG downloading tarball https://github.com/client9/misspell/releases/download/v0.3.4/misspell_0.3.4_linux_64bit.tar.gz
client9/misspell DEBUG downloading checksum https://github.com/client9/misspell/releases/download/v0.3.4/misspell_0.3.4_checksums.txt
client9/misspell INFO installed as /tmp/tmp.S9YRR5wJWm/misspell
Please fix the listed misspell errors and verify using 'make misspell'
make: *** [Makefile:184: misspell] Error 1
```

## Expected Behavior

I'd like to still know where my misspellings (which are frequent) are happening so I can correct them. Maybe the output is getting eaten up somewhere?

## Steps to Reproduce the Bug

1. Add a misspelling somewhere
2. See `make misspell` not output the location of the misspelling

--- a/.github/workflows/check-all.yaml
+++ b/.github/workflows/check-all.yaml
@@ -1,4 +1,4 @@
-name: Check - Main (All tests)
+name: Check - All linters, etc
 on:
   push:
     branches:
@@ -43,6 +43,7 @@ jobs:
       - name: Run all checks
         shell: bash
         run: |
+          make ensure-deps
           make check
       - name: Get dependencies
         shell: bash

--- a/Makefile
+++ b/Makefile
@@ -82,11 +82,13 @@ ifdef GOBIN
 TANZU_CLI_INSTALL_PATH = "$${GOBIN}/tanzu"
 endif
 
+NUMBER_OF_CORES ?= 1
 ifeq ($(TCE_CI_BUILD), true)
 XDG_DATA_HOME := /tmp/mylocal
 XDG_CACHE_HOME := /tmp/mycache
 XDG_CONFIG_HOME := /tmp/myconfig
 SED := sed -i
+NUMBER_OF_CORES := $(shell nproc --all)
 endif
 ifeq ($(XDG_DATA_HOME),)
 ifeq ($(build_OS), Darwin)
@@ -94,6 +96,7 @@ XDG_DATA_HOME := "${HOME}/Library/Application Support"
 XDG_CACHE_HOME := ${HOME}/.cache
 XDG_CONFIG_HOME := ${HOME}/.config
 SED := sed -i '' -e
+NUMBER_OF_CORES := $(shell sysctl -n hw.physicalcpu)
 endif
 endif
 ifeq ($(XDG_DATA_HOME),)
@@ -102,6 +105,7 @@ XDG_DATA_HOME := ${HOME}/.local/share
 XDG_CACHE_HOME := ${HOME}/.cache
 XDG_CONFIG_HOME := ${HOME}/.config
 SED := sed -i
+NUMBER_OF_CORES := $(shell nproc --all)
 endif
 endif
 
@@ -118,8 +122,24 @@ OCI_REGISTRY := projects.registry.vmware.com/tce
 ##### IMAGE
 
 ##### LINTING TARGETS
-.PHONY: lint mdlint shellcheck check yamllint misspell actionlint urllint imagelint
-check: ensure-deps lint mdlint shellcheck yamllint misspell actionlint urllint imagelint
+.PHONY: check check-serial check-parallel lint mdlint shellcheck yamllint misspell actionlint urllint imagelint
+check:
+	@make_version=$(shell make --version | grep -E "GNU Make [0-9].[0-9]+" | grep -E -o "[0-9].[0-9]+" | cut -d "." -f1); \
+	if [ "$${make_version}" != "3" ]; then \
+		echo " "; \
+		echo "*********************************************************************************"; \
+		echo "Executing linters in parallel. Output is withheld until individual lint jobs complete."; \
+		echo "*********************************************************************************"; \
+		echo " "; \
+		$(MAKE) --jobs=$(NUMBER_OF_CORES) --output-sync=target lint mdlint shellcheck yamllint misspell actionlint urllint imagelint; \
+	else \
+		echo " "; \
+		echo "***************************************************"; \
+		echo "To speed up linting, update to the latest GNU Make."; \
+		echo "***************************************************"; \
+		echo " "; \
+		$(MAKE) lint mdlint shellcheck yamllint misspell actionlint urllint imagelint; \
+	fi;
 
 .PHONY: check-deps-minimum-build
 check-deps-minimum-build:

--- a/hack/check/.misspellignore
+++ b/hack/check/.misspellignore
@@ -1,3 +1,4 @@
 go.mod
 go.sum
 addons.*upstream.*
+.git

--- a/hack/check/check-misspell.sh
+++ b/hack/check/check-misspell.sh
@@ -21,15 +21,15 @@ misspellignore_files="${CHECK_DIR}/.misspellignore"
 ignore_files=$(cat "${misspellignore_files}")
 
 # check spelling
-RET_VAL=$(git ls-files | grep -v "${ignore_files}" | xargs "${MISSPELL_DIR}/misspell" | grep "misspelling")
+FILES_TO_CHECK=$(git ls-files | grep -v "${ignore_files}")
+echo " "
+echo " "
+echo "Files to check spelling:"
+echo "${FILES_TO_CHECK}"
+echo " "
+echo " "
+echo "Errors:"
+git ls-files | grep -v "${ignore_files}" | xargs "${MISSPELL_DIR}/misspell"
 
 # delete the directory to return environment to original condition
 rm -rf "${MISSPELL_DIR}"
-
-# check return value
-if [[ "${RET_VAL}" != "" ]]; then
-  echo "Please fix the listed misspell errors and verify using 'make misspell'"
-  exit 1
-else
-  echo "misspell check passed!"
-fi

